### ISSUE TITLE
Sensitive data redaction support in libsinsp 

### ISSUE THREAD

**Issue by 0snug0** [falcosecurity/libs#17]
**Motivation**
When capturing syscalls, we capture all data including sensitive information that should be removed prior to forwarding to a central registry. There is no filter to exclude or redact passwords, secrets and other sensitive data from being saved. The data should be removed at the source.

**Feature**
Regex capabilities to match sensitive sensitive data
Capabilities to redact (replace with ***) or remove the sensitive data

**Alternatives**
Considered removing the sensitive data after write - not secure enough

**Additional context**

**Comment by FedeDP** [comment]
Hi @0snug0 ! 
i agree with you, we somewhat need this.
I think we can try to give this a shoot for the 0.32 release cycle.

Thanks for the input!

/remove-lifecycle stale

**Comment by incertum** [comment]
First of all @0snug0 yes this is very relevant from a data privacy point of view.

These are the technical hurdles:

Generic regex matching capabilities based on end user input is not yet supported and it may be a bigger ask than it sounds, because of possible performance impacts.

Even if we did already support above, the end user may still encounter significant challenges when attempting to correctly redact logs when applicable as applications constantly change.

By the way do you have concrete examples? For the most part it may be rare cases that the clear text data in question makes it to one the field outputs?

In summary, I do not yet have a good idea on how to support this feature in a sustainable way. Open to hearing more suggestions from the community.
